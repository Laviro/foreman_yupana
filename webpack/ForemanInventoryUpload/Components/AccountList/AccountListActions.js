import API from 'foremanReact/API';
import { inventoryUrl } from '../../ForemanInventoryHelpers';
import {
  INVENTORY_ACCOUNT_STATUS_POLLING,
  INVENTORY_ACCOUNT_STATUS_POLLING_ERROR,
  INVENTORY_ACCOUNT_STATUS_POLLING_START,
  INVENTORY_ACCOUNT_STATUS_POLLING_STOP,
  INVENTORY_PROCESS_RESTART,
} from './AccountListConstants';

export const fetchAccountsStatus = () => async dispatch => {
  try {
    const {
      data: { accounts, autoUploadEnabled, hostObfuscationEnabled },
    } = await API.get(inventoryUrl('accounts'));
    dispatch({
      type: INVENTORY_ACCOUNT_STATUS_POLLING,
      payload: {
        accounts,
        autoUploadEnabled,
        hostObfuscationEnabled,
      },
    });
  } catch (error) {
    dispatch({
      type: INVENTORY_ACCOUNT_STATUS_POLLING_ERROR,
      payload: {
        error: error.message,
      },
    });
  }
};

export const startAccountStatusPolling = pollingProcessID => ({
  type: INVENTORY_ACCOUNT_STATUS_POLLING_START,
  payload: {
    pollingProcessID,
  },
});

export const stopAccountStatusPolling = pollingProcessID => dispatch => {
  clearInterval(pollingProcessID);
  dispatch({
    type: INVENTORY_ACCOUNT_STATUS_POLLING_STOP,
  });
};

export const restartProcess = (accountID, activeTab) => dispatch => {
  let processController = null;
  let processStatusName = null;

  if (activeTab === 'uploading') {
    processController = 'uploads';
    processStatusName = 'upload_report_status';
  } else {
    processController = 'reports';
    processStatusName = 'generate_report_status';
  }

  API.post(inventoryUrl(`${accountID}/${processController}`));
  dispatch({
    type: INVENTORY_PROCESS_RESTART,
    payload: {
      accountID,
      processStatusName,
    },
  });
};
