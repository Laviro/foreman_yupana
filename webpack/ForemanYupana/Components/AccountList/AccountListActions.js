import { ACCOUNTLIST_CHANGE_BOOL } from './AccountListConstants';

export const changeBool = resource => dispatch => {
  dispatch({
    type: ACCOUNTLIST_CHANGE_BOOL,
    payload: resource,
  });
};
