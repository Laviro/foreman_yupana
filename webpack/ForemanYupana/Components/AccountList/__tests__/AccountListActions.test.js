import { testActionSnapshotWithFixtures } from 'react-redux-test-utils';
import { changeBool } from '../AccountListActions';

const fixtures = {
  'should changeBool': () => changeBool({ bool: true }),
};

describe('AccountList actions', () => testActionSnapshotWithFixtures(fixtures));
