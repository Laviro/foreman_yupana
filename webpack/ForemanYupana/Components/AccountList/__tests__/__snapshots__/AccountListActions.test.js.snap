// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountList actions should changeBool 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "bool": true,
      },
      "type": "ACCOUNTLIST_CHANGE_BOOL",
    },
  ],
]
`;
